src/topology/algebra/infinite_sum.lean

```diff
@@ -891,30 +938,6 @@ lemma tsum_comm [regular_space α] {f : β → γ → α} (h : summable (functio
   ∑' c b, f b c = ∑' b c, f b c :=
 tsum_comm' h h.prod_factor h.prod_symm.prod_factor
 
-/-- Let `f : ℕ → ℝ` be a sequence with summable series and let `i ∈ ℕ` be an index.
-Lemma `tsum_ite_eq_extract` writes `Σ f n` as the sum of `f i` plus the series of the
-remaining terms. -/
-lemma tsum_ite_eq_extract {f : ℕ → ℝ} (hf : summable f) (i : ℕ) :
-  ∑' n, f n = f i + ∑' n, ite (n = i) 0 (f n) :=
-begin
-  refine ((tsum_congr _).trans $ tsum_add (hf.summable_of_eq_zero_or_self _) $
-    hf.summable_of_eq_zero_or_self _).trans (add_right_cancel_iff.mpr (tsum_ite_eq i (f i)));
-  exact λ j, by { by_cases ji : j = i; simp [ji] }
-end
```

Comment:
I've removed this lemma because:
* my proof doesn't need it and so it is no longer used
* generalizing it results in a messy error
* the term-mode part of this proof makes it very hard to diagnose the problem, so it's easier just to throw it out.

Reply:
My only reason for keeping this lemma is that it does not use existence of additive inverses in the the target.  For this reason, it would apply also to `nnreal`, for instance, even though it is not stated in that case.

Reply:
I agree, it would be nice to have a lemma that does not require additive inverses - however, in its current form this lemma is not allow that anyway, and I couldn't find the set of typeclasses needed to make the existing proof work.

Reply:
I see, I did not realize that the assumption on \a is that it is a group!  Still, it would be nice if the statement `summable.summable_of_eq_zero_or_self` did not require it, but I am not even sure if it is true without assuming existence of opposites.

I am a little skeptical about removing the lemma, since sometimes it is useful to extract a single (or maybe a finite number of) term(s) of a series.

Reply:
Is it possible to leave the lemma as-is with a TODO to generalize?

Reply:
Sure, I could just add it back unchanged:
```suggestion
/-- Let `f : ℕ → ℝ` be a sequence with summable series and let `i ∈ ℕ` be an index.
Lemma `tsum_ite_eq_extract` writes `Σ f n` as the sum of `f i` plus the series of the
remaining terms.

TODO: generalize this to `f : β→ α` with appropriate assumptions
-/
lemma tsum_ite_eq_extract {f : ℕ → ℝ} (hf : summable f) (i : ℕ) :
  ∑' n, f n = f i + ∑' n, ite (n = i) 0 (f n) :=
begin
  refine ((tsum_congr _).trans $ tsum_add (hf.summable_of_eq_zero_or_self _) $
    hf.summable_of_eq_zero_or_self _).trans (add_right_cancel_iff.mpr (tsum_ite_eq i (f i)));
  exact λ j, by { by_cases ji : j = i; simp [ji] }
end
```